src/Fantomas.Core/CodePrinter.fs

```diff
@@ -2088,215 +2088,12 @@ and genExpr astContext synExpr ctx =
 
             expressionFitsOnRestOfLine short long
 
+        | DotGetAppWithLambda ((e, es, lpr, lambda, rpr, pr), lids) ->
+            genAppWithLambda (e, es, lpr, lambda, rpr, pr) false (Some lids) astContext
```

Comment:
I wouldn't pass in the `lids` of the `DotGet` here to the `genAppWithLambda` function.
What you want to control from outside the `genAppWithLambda` is whether there should be a space or not.

I would pass in a function (that adds the space) instead of a boolean.

Something long the lines of:

```
        | DotGetAppWithLambda ((e, es, lpr, lambda, rpr, pr), lids) ->
            leadingExpressionIsMultiline 
                (genAppWithLambda astContext sepNone (e, es, lpr, lambda, rpr, pr))
                (fun isMultline -> // print lids and do the right thing based on isMultline)

        // functionName arg1 arg2 (fun x y z -> ...)
        | AppWithLambda (e, es, lpr, lambda, rpr, pr) ->
              let sepSpaceAfterFunctionName =
                  let sepSpaceBasedOnSetting e =
                      match e with
                      | Paren _ -> sepSpace
                      | UppercaseSynExpr -> (fun ctx -> onlyIf ctx.Config.SpaceBeforeUppercaseInvocation sepSpace ctx)
                      | LowercaseSynExpr -> (fun ctx -> onlyIf ctx.Config.SpaceBeforeLowercaseInvocation sepSpace ctx)

                  match es with
                  | [] -> sepSpaceBasedOnSetting e
                  | _ -> sepSpace

            genAppWithLambda astContext sepSpaceAfterFunctionName  (e, es, lpr, lambda, rpr, pr) 
```

The benefit here is that `genAppWithLambda` has no knowledge about `lids` so the function only does one thing (namely printing a function application that ends with a lambda).

Oh and in a lot of places `astContext` is the first argument.
This is a bit historically.

Reply:
Thanks for the review and the suggestions. Really appreciate your patience.

I hope I'm not too confused here but I think this is now more in line with our default config of `SpaceBeforeUppercaseInvocation = false`. As a consequence there were some tests which needed adjustments.

Reply:
I think the changes in the existing tests are all valid.